demos/common/python/openvino/model_zoo/model_api/README.md

```diff
@@ -0,0 +1,150 @@
+| Semantic Segmentation      |  <ul><li>`SegmentationModel`</li></ul> |
+
+## Model API Adapters
+
+Model API wrappers are executor-agnostic, meaning it does not implement the specific model inference or model loading, instead it can be used with different executors having the implementation of common interface methods in adapter class respectively.
+
+Currently, `OpenvinoAdapter` and `OVMSAdapter` are supported.
+
+#### OpenVINO Adapter
+
+`OpenvinoAdapter` hides the OpenVINO™ toolkit API, which allows Model API wrappers launching with models represented in Intermediate Representation (IR) format.
+It accepts a path to either `xml` model file or `onnx` model file.
+
+#### OpenVINO Model Server Adapter
+
+`OVMSAdapter` hides the OpenVINO Model Server python client API, which allows Model API wrappers launching with models served by OVMS.
+
+Refer to __[`OVMSAdapter`](adapters/ovms_adapter.md)__ to learn about running demos with OVMS.
+
+For OpenVINO Model Server Adapter employment, you need to install the package with extra module:
+```sh
+pip install <omz_dir>/demos/common/python[ovms]
+```
+
+## Model API Pipelines
+
+Model API Pipelines represent the high-level wrappers upon the input data and accessing model results management.
+It performs the data submission for model inference, verification of inference status, whether the result is ready or not, and results accessing.
+
+The `AsyncPipeline` is available, which handles the asynchronous execution of a single model.
+
+## Ready-to-use Model API solutions
+
+To apply Model API wrappers in custom applications, learn the provided example of common scenario of how to use Python* Model API.
+
+ In the example, the SSD architecture is used to predict bounding boxes on input image `"sample.png"`. The model execution is produced by `OpenvinoAdapter`, therefore we submit the path to the model's `xml` file.
+
+Once the SSD model wrapper instance is created, we get the predictions by the model in one line: `ssd_model(input_data)` - the wrapper performs the preprocess method, synchronous inference on OpenVINO™ toolkit side and postprocess method.
+
+```python
+import cv2
+# import model wrapper class
+from openvino.model_zoo.model_api.models import SSD
+# import inference adapter and helper for runtime setup
+from openvino.model_zoo.model_api.adapters import OpenvinoAdapter, create_core
+
+
+# read input image using opencv
+input_data = cv2.imread("sample.png")
+
+# define the path to mobilenet-ssd model in IR format
+model_path = "public/mobilenet-ssd/FP32/mobilenet-ssd.xml"
+
+# create adapter for OpenVINO™ runtime, pass the model path
+model_adapter = OpenvinoAdapter(create_core(), model_path, device="CPU")
+
+# create model API wrapper for SSD architecture
+# preload=True loads the model on CPU inside the adapter
+ssd_model = SSD(model_adapter, preload=True)
+
+# apply input preprocessing, sync inference, model output postprocessing
+results = ssd_model(input_data)
+```
```

Comment:
We won't ever verify the snippet works. To me it's the strong reason to delete it. You should refer to a demo instead

Reply:
If the majority prefers to delete it, I will do it

Reply:
This is not about voting.
1. This is about explaining why your solution is correct.
2. Even if you insist on conducting a poll,  your voice won't count. My voice won't count either. Only the voice of the person who is responsible for OMZ matters. Which is @vladimir-dudnik's one

Reply:
My point is the demos contain only complex cases with asynchronous models execution, it is not the only one usage scenario. We should provide somewhere the example of simple synchronous model call.

Didn't get point about the snippet work verification. Many packages have a documentation with API examples. The package is going to be updated with new releases, and with the new releases the documentation will be also updated as well as the snippet.

Reply:
You can write a dedicated sample and refer to it. You will anyway need to cover your lib with tests. The sample could be a part of tests.

> Didn't get point about the snippet work verification. Many packages have a documentation with API examples. The package is going to be updated with new releases, and with the new releases the documentation will be also updated as well as the snippet.

Such packages have people whos work to continuously check the example is consistent. We don't have such people. If packages don't do that, they usually end up having broken examples.

Reply:
Here is the solution. Keep this section, but remove it after tests are added

Reply:
Okay